### ISSUE TITLE
Porters Handbook in Chinese

### ISSUE THREAD

**Issue by engerim42** [FreeBSDTaiwan/portscamp#2]
Looking at https://www.freebsd.org/doc/en/books/porters-handbook/
vs. https://www.freebsd.org/doc/zh_TW/books/porters-handbook/
The zh_TW version is indeed horribly outdated (2008).

Given that only a small amount of the EN guide was translated, it would make sense to start this from scratch.

**Comment by engerim42** [comment]
That said, it would probably make sense to do it Chapter by Chapter during each meetup.

**Comment by mose** [comment]
Yes I think the translation efforts would be a separate trend, we will see who is showing some enthusiasm about it. The FreeBSD foundation also told us they have various other material that are not up to date, they are going to provide us a list.

**Comment by mose** [comment]
Here is a comment from Ray about it:

> The Porters' Handbook in Traditional Chinese is outdated. And the
> document and file structure are outdated, too. We need to update it.
> I've contacted The FreeBSD Documentation Engineering Team (wblock). He
> suggested to use the PO translation system for any new translation work.
> I will shift the translation work to PO translation recently. But after
> the shift, the previous tranditional Chinese translation work will lost
> and if some of the previous work are useful, we need copy and paste
> manually. I will backup the previous xml file before the shift work. If
> you need the previous translation, you could check the revision 47831
> out of the freebsd-doc repository:
> 
> svn checkout -r 47831  https://svn.FreeBSD.org/doc/head/zh_TW.UTF-8/books/porters-handbook /tmp/porters-handbook
> 
> Best Regards,
> 
> RayCherng

**Comment by engerim42** [comment]
Found the discussions in freebsd-translators in [October](https://lists.freebsd.org/pipermail/freebsd-translators/2015-October/thread.html) and [December](https://lists.freebsd.org/pipermail/freebsd-translators/2015-December/thread.html) archives.
Ultimately this leads to [this howto](https://www.freebsd.org/doc/en_US.ISO8859-1/books/fdp-primer/po-translations-creating.html).
I'd suggest we install [weblate](https://github.com/nijel/weblate/) on a PaaS or IaaS instance and let translators feed it. This will allow us to do reviews and maybe integration.
And in case its not obvious, we should use UTF-8 instead of [current](http://ftp.freebsd.org/pub/FreeBSD/doc/zh_TW.Big5/books/porters-handbook/) Big-5 encoded version.

**Comment by raycherng** [comment]
We could use a web-based translation system like Pootle.
PC-BSD uses Pootle
https://lists.freebsd.org/pipermail/freebsd-translators/2015-October/000042.html

lwhsu has converted the Big5 to UTF-8 last year.
http://ftp.freebsd.org/pub/FreeBSD/doc/zh_TW.UTF-8/books/porters-handbook/

**Comment by engerim42** [comment]
Both Pootle or weblate should be fine. I prefer the weblate interface+features.
Here is what I'd like to do this week (I'm on vacation) :+1: 
1. Set up a VPS + weblate server for us
2. A script that fetches the SVN of the porters handbook (EN version, possibly once per week) and executes `make po` as per [Example 12.1 here](https://www.freebsd.org/doc/en/books/fdp-primer/po-translations-creating.html) . Then continuously imports this to weblate.
3. Make a script that fetches the PO file from git

**Comment by c9s** [comment]
cc @pct
